holidays/financial/us_new_york.py

```diff
@@ -0,0 +1,162 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+
+from holidays.calendars.gregorian import MAY, JUL, MON, TUE, WED, THU, FRI, SAT, SUN
+from holidays.constants import HALF_DAY, PUBLIC
+from holidays.groups import ChristianHolidays, InternationalHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    ObservedRule,
+    SAT_TO_PREV_FRI,
+    SUN_TO_NEXT_MON,
+)
+
+SIFMA_EARLY_CLOSE = ObservedRule({MON: -3, TUE: -1, WED: -1, THU: -1, FRI: -1, SAT: -2, SUN: -2})
+
+
+class USNewYork(ObservedHolidayBase, ChristianHolidays, InternationalHolidays):
+    """US New York (USNY) financial market holidays.
+
+    The USNY calendar represents general "New York business days" used in financial contracts
+    (ISDA derivatives, OTC instruments). While it follows SIFMA (Securities Industry and
+    Financial Markets Association) recommendations for most holidays, it differs from both
+    USGS (US Government Securities) and NYSE calendars.
+
+    Key differences:
+        - vs USGS: USNY does NOT observe Good Friday (USGS does)
+        - vs NYSE: USNY DOES observe Columbus Day and Veterans Day (NYSE does not in modern times)
+        - vs NYSE: USNY does NOT include special one-off closures (NYSE does)
```

Comment:
As I can see, USNY is actually New York Banking Holidays, and don't follow SIFMA. If you know of sources that confirm otherwise, please cite them.